test/powershell/Modules/Microsoft.PowerShell.Utility/RunspaceCmdlets.Tests.ps1

```diff
@@ -0,0 +1,39 @@
+Describe "Get-Runspace cmdlet tests" -Tag "CI" {
+    BeforeAll {
+        # Clear out all unexpected runspaces
+        # id 1 is the one we started with
+        Get-Runspace |?{$_.id -ne 1} | %{$_.dispose()}
+        $CurrentRunspace = $ExecutionContext.Host.Runspace
+        $ExpectedInstanceId = $CurrentRunspace.InstanceId
+        $ExpectedId = $currentRunspace.Id
+    }
+    It "Get-Runspace should return the current runspace" {
+        $runspace = get-runspace
+        $runspace.InstanceId | Should be $ExpectedInstanceId
+    }
+    It "Get-Runspace with runspace InstanceId should return the correct runspace" {
+        $runspace = get-runspace -instanceid $CurrentRunspace.InstanceId
+        $runspace.InstanceId | Should be $ExpectedInstanceId
+    }
+    It "Get-Runspace with runspace Id should return the correct runspace" {
+        $runspace = get-runspace -id $CurrentRunspace.Id
+        $runspace.InstanceId | Should be $ExpectedInstanceId
+    }        
+    Context "Multiple Runspaces" {
+        BeforeAll {
+            $runspaceCount = @(get-runspace).count
+            $r1 = [runspacefactory]::CreateRunspace()
+            $r1.Open()
+            $r2 = [runspacefactory]::CreateRunspace()
+        }
+        AfterAll {
+            $r1.Close()
+            $r1.Dispose()
+            $r2.Dispose()
+        }
+        It "Get-Runspace should return all runspaces" {
+            $expectedCount = $runspaceCount + 2
+            (get-runspace).Count | should be $expectedCount
```

Comment:
there's no contemplation of running tests in parallel _in the same process_, which is where there would be a problem (and probably with a very large number of our other tests). Running tests in parallel processes won't affect this test at all, however the point is taken

---
In reply to: [78447130](https://github.com/PowerShell/PowerShell/pull/2239#discussion_r78447130) [](ancestors = 78447130)